### ISSUE TITLE
Request for option to suppress warning associated with #1

### ISSUE THREAD

**Issue by heafnerj** [ho-tex/hypdoc#4]
May I please request a package option that suppresses the warning associated with [ho-tex/hypdoc/issues/1](https://github.com/ho-tex/hypdoc/issues/1 ) related to the loading of `hyperref`? Give that `hypdoc` is loaded by `doc` and `ltxdoc` I don't know if this is possible. I understand that the warning is harmless but it would be nice to not have any warnings in general use when a document is successfully "compiled."

**Comment by u-fischer** [comment]
Well hypdoc can't know if the compilation was successful. If you loaded  hyperref previously with different options something could be missing.  The best would be if you wouldn't load hyperref manually.

**Comment by heafnerj** [comment]
I am not loading `hyperref` manually. It is being loaded by `tcolorbox`'s `documentation` library. I need `hypdoc` to make page numbers in my Change History hyperlinks. Perhaps just changing the message text so it doesn't show up in the log or console as an actual "Warning" would be easier.

**Comment by davidcarlisle** [comment]
medium (or even short) term, if we move hyperref to the new package option handler in the format, option clashes should mostly be a thing of the past.

**Comment by u-fischer** [comment]
> medium (or even short) term, if we move hyperref to the new package option handler in the format, option clashes should mostly be a thing of the past.

Yes, but it wouldn't solve this problem: you would then get warnings from the keyval handler that there is a try to load hyperref again with a load-only option (hyperindex=false).

Imho the best here currently is to load tcolorbox later. 

~~~~
\AddToHook{begindocument/before}{\usepackage[listings,documentation]{tcolorbox}} 
~~~~

**Comment by heafnerj** [comment]
I was wondering whether or not a hook solution might be appropriate, but I wasn't sure where it would have to go. I've been studying the source code for `dot/ltxdoc`, `hypdoc`, and `tcolorbox`'s `documentation` and the interplay is rather complex.

**Comment by muzimuzhi** [comment]
Or redirect that warning to info, rather than suppress it.

**Comment by muzimuzhi** [comment]
> May I please request a package option that suppresses the warning associated with [ho-tex/hypdoc/issues/1](https://github.com/ho-tex/hypdoc/issues/1) related to the loading of `hyperref`?

@heafnerj 
- Currently: a package warning is raised
- Your (original) proposal: a package option to suppress the warning
- Mine: a package option to redirect warning to info (`\PackageInfo{<pkg>}{<msg>}`) which is ignored by some LaTeX editors' log analyzers.

**Comment by heafnerj** [comment]
Ah yes. That's precisely what I meant in the last sentence of my initial post. I didn't know the correct terms to use. That would certainly address my concern.
